src/main/scala/com/webtrends/harness/component/akkahttp/client/SimpleHttpClient.scala

```diff
@@ -0,0 +1,46 @@
+package com.webtrends.harness.component.akkahttp.client
+
+import akka.actor.Actor
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{HttpMethods, HttpRequest, HttpResponse, StatusCodes}
+import akka.stream.ActorMaterializer
+import akka.util.ByteString
+import com.webtrends.harness.logging.ActorLoggingAdapter
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait SimpleHttpClient extends ActorLoggingAdapter { this: Actor =>
+
+  implicit val executionContext: ExecutionContext
+  implicit val materializer = ActorMaterializer()
+
+  private val http = Http(context.system)
+
+  def request(req: HttpRequest): Future[(HttpResponse, Array[Byte])] = {
+    http.singleRequest(req).flatMap { response =>
+      response.entity.dataBytes.runFold(ByteString(""))(_ ++ _).map(bs => (response, bs.toArray))
```

Comment:
So the default size of a HTTP response is set to 8MB by akka http. It would fail here if you tried to request something that returned a response > 8MB. You could use `response.entity.withoutSizeLimit` or expose the size limit as a param to `request` method

Reply:
Akka HTTP has already has an Unmarshaller for byte arrays. You could use that here instead of manually creating the byte array: `PredefinedFromEntityUnmarshallers..byteArrayUnmarshaller`. You can use it like this: 

```scala

implicit private val um
  : Unmarshaller[HttpEntity, Array[Byte]] = PredefinedFromEntityUnmarshallers
    .byteArrayUnmarshaller
http.singleRequest(req).flatMap { res => Unmarshal(res.entity.withoutSizeLimit()).to[Array[Byte]])

```